packages/lit-element/src/lib/platform-support.ts

```diff
@@ -0,0 +1,388 @@
+  _createElement(html: string): HTMLTemplateElement;
+}
+
+/**
+ * lit-html patches. These properties cannot be renamed.
+ * * NodePart.prototype._getTemplate
+ * * NodePart.prototype._setValue
+ */
+(globalThis as any)['litHtmlPlatformSupport'] = ({
+  NodePart,
+  Template,
+}: {
+  NodePart: NodePartStandIn;
+  Template: TemplateStandIn;
+}) => {
+  if (!needsPlatformSupport) {
+    return;
+  }
+
+  // console.log(
+  //   '%c Making lit-html compatible with ShadyDOM/CSS.',
+  //   'color: lightgreen; font-style: italic'
+  // );
+
+  const styledScopes: Set<string> = new Set();
+  const scopeCssStore: Map<string, string[]> = new Map();
+  let currentScope: string;
+
+  const isScopeStyled = (name: string | undefined) =>
+    needsPlatformSupport && name !== undefined ? styledScopes.has(name) : true;
+
+  const cssForScope = (name: string) => {
+    if (isScopeStyled(name)) {
+      return;
+    }
+    let scopeCss = scopeCssStore.get(name);
+    if (scopeCss === undefined) {
+      scopeCssStore.set(name, (scopeCss = []));
+    }
+    return scopeCss;
+  };
+
+  const ensureStyles = (name: string, template: HTMLTemplateElement) => {
+    if (isScopeStyled(name)) {
+      return;
+    }
+    // Mark this scope as styled.
+    styledScopes.add(name);
+    // Remove stored data
+    scopeCssStore.delete(name);
+    window.ShadyCSS!.prepareTemplateStyles(template, name);
+  };
+
+  const scopedTemplateCache = new Map<
+    string,
+    Map<TemplateStringsArray, TemplateStandIn>
+  >();
+
+  // Note, it's ok to subclass Template since it's only used via NodePart.
+  let currentResult: ShadyTemplateResult;
+  class ShadyTemplate extends Template {
+    constructor(result: ShadyTemplateResult) {
+      // TODO(sorvell): Decide if Template should store the result to avoid this.
+      // We need the `result` in `_createElement` which is called directly
+      // from the constructor. Since we can't access this before super, uce
+      // a side channel to store the data.
+      currentResult = result;
+      super(result);
+    }
+    /**
+     * Override to extract style elements from the template
+     * and store all style.textContent in the shady scope data.
+     */
+    _createElement(html: string) {
+      const template = super._createElement(html);
+      if (!window.ShadyCSS!.nativeShadow) {
+        window.ShadyCSS!.prepareTemplateDom(template, currentScope);
+      }
+      const scopeCss = cssForScope(currentScope);
+      if (scopeCss !== undefined) {
+        // Remove styles and store textContent.
+        const styles = template.content.querySelectorAll('style') as NodeListOf<
+          HTMLStyleElement
+        >;
+        scopeCss.push(
+          ...Array.from(styles).map((style) => {
+            style.parentNode?.removeChild(style);
+            return style.textContent!;
+          })
+        );
+        // Apply styles if this is the result rendering into shadowRoot.
+        if (currentResult.__renderIntoShadowRoot) {
```

Comment:
If we're not rendering into shadowRoot, then what? Shouldn't we not touch anything if so?

Reply:
I see, we _accumulate_ if we have a scope but are not the root template going into the SR.